packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -454,7 +470,9 @@ class MLKitScannerPageState extends LifecycleAwareState<MLKitScannerPage>
     CameraHelper.destroyControllerInstance();
 
     await _streamSubscription?.cancel();
-    await _barcodeDecoder.onDispose();
+    try {
+      await _barcodeDecoder.onDispose().timeout(const Duration(seconds: 5));
```

Comment:
Perhaps, as is already done with `_controller`, the `onDispose` method could be done without `await`ing by creating each time a new `_barcodeDecoder`.

Reply:
The `await` is required to ensure everything is closed correctly (we already have lots of concurrency issues).
It's only on certain cases where this method can take a long time and instead of waiting for ages, we let the process continue, to allow the user to have the camera feed

Reply:
I understand the general purpose of `await`.
Actually I assumed that you could create a new instance of "barcode decoder" when you needed it, which would have given more flexibility. Like, "take your time to dispose it, I don't need to wait, anyway I'll take a brand new one next time".
But that doesn't seem to be the case if I read the code more carefully, that doesn't seem to be a possible quick change, and I'm not even sure that would help fixing the issue.

As long as your code fixes the issue, that's good for me!